src/Support/ErrorMessage.hpp

```diff
@@ -0,0 +1,43 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===-------------------------- ErrorMessage.hpp --------------------------===//
+//
+// Copyright 2019-2020 The IBM Research Authors.
+//
+// =============================================================================
+//
+// Functions for adding passes and processing input files.
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <errno.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+//
+// Return error message for given error number for opening file at given path
+// If error number is not given, investigate it by opening it.
+//
+#define PATH_SIZE 4096
+static std::string getErrorMessageforFileOpeningErrors(
```

Comment:
I wonder, is there no way to use llvm for this? We can leverage any of the existing llvm headers/libraries and I would expect that there's something similar there.

Reply:
To support Windows environment, I use llvm::sys::StrError in this function to get error string from error number.
I hope that it works for WIndows.

Reply:
I don't think that will work for windows, especially since the APIs are different.

I believe llvm has functionality already defined to perform a lot of file operations, possibly including the exact thing you are doing. Have you looked to see if it can be re-used?